cmd/generate.go

```diff
@@ -157,31 +163,67 @@ function fetch_bench_io_results() {
 
 # Run Netperf benchmark
 function bench_net() {
-  server=$(roachprod ip "$CLUSTER":4)
+  if [ $NODES -lt 2 ]
+  then
+    echo "NODES must be greater than 1 for this test"
+    exit 1
+  fi
+
+  server=$(roachprod ip "$CLUSTER":$NODES)
   port=1337
   # Start server
-  roachprod run "$CLUSTER":4 ./scripts/gen/network-netperf.sh -- -S -p $port
+  roachprod run "$CLUSTER":$NODES ./scripts/gen/network-netperf.sh -- -S -p $port
 
   # Start client
-  run_under_tmux "net" "$CLUSTER:3" "./scripts/gen/network-netperf.sh -s $server -p $port $net_extra_args"
+  run_under_tmux "net" "$CLUSTER:$((NODES-1))" "./scripts/gen/network-netperf.sh -s $server -p $port $net_extra_args"
 }
 
 # Wait for Netperf benchmark to complete and fetch results.
 function fetch_bench_net_results() {
-  roachprod run "$CLUSTER":3 ./scripts/gen/network-netperf.sh -- -w
-  roachprod get "$CLUSTER":3 ./netperf-results $(results_dir "netperf-results")	
+  if [ $NODES -lt 2 ]
+  then
+    echo "NODES must be greater than 1 for this test"
+    exit 1
+  fi
+
+  if [ $NODES -eq 2 ]
+  then
+    roachprod run "$CLUSTER":1 ./scripts/gen/network-netperf.sh -- -w
+    roachprod get "$CLUSTER":1 ./netperf-results $(results_dir "netperf-results")
+  else
+    roachprod run "$CLUSTER":3 ./scripts/gen/network-netperf.sh -- -w
+    roachprod get "$CLUSTER":3 ./netperf-results $(results_dir "netperf-results")
+  fi
 }
 
 # Run TPCC Benchmark
 function bench_tpcc() {
- start_cockroach
- pgurls=$(roachprod pgurl "$CLUSTER":1-$((NODES-1)))
- run_under_tmux "tpcc" "$CLUSTER:4" "./scripts/gen/tpcc.sh $tpcc_extra_args ${pgurls[@]}"
+  if [ $NODES -lt 2 ]
+  then
+    echo "NODES must be greater than 1 for this test"
+    exit 1
+  fi
+
+  start_cockroach
+  if [ $NODES -eq 2 ]
+  then
+    pgurls=$(roachprod pgurl "$CLUSTER":1)
```

Comment:
I'm pretty sure the if-branch is equivalent to the else branch, as `$CLUSTER:1` is equal to `$CLUSTER:1-1`. You can verify this on the command line:

```
~ roachprod pgurl local:1
'postgres://root@127.0.0.1:26257?sslmode=disable'
~ roachprod pgurl local:1-1
'postgres://root@127.0.0.1:26257?sslmode=disable'
```